flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -1256,7 +1256,10 @@ void checkTaskManagerTimeoutsAndRedundancy() {
 			}
 
 			int slotsDiff = redundantTaskManagerNum * numSlotsPerWorker - freeSlots.size();
-			if (slotsDiff > 0) {
+			if (freeSlots.size() == slots.size()) {
```

Comment:
If job's failover takes a long time, we could run into this scenario as well. In that case, the redundant TM mechanism will not take effect as expected. WDYT about this?

cc @tillrohrmann

Reply:
I think in a first version this should be good enough. Later one could ask the `ResourceManager.jobLeaderIdService` if there is still a job registered. However, this service also times out the job if no leader registers within a given timeout.

Reply:
Ok, that makes sense to me.

Reply:
If the job failover takes long, then all the idle task managers will timeout and be released. There's few benefit keeping the redundant task managers, since we need to wait for re-launching the other task managers anyway. (Assuming the job needs more task managers than the redundant ones to execute.)